plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -148,6 +152,8 @@ public class MongoSession
     private final Cache<SchemaTableName, MongoTable> tableCache;
     private final String implicitPrefix;
 
+    private final ConnectorExpressionRewriter<Document> connectorExpressionRewriter;
```

Comment:
Can we inject this `connectorExpressionRewriter` as a part of MongoMetadata ?

Reply:
I saw the codes of `DefaultJdbcMetadata` are invoked `jdbcClient.convertPredicate()` function to generate expression pushdown, not handling expression rewrite itself.

The MongoSession class is like JdbcClient which encapsulates the logic of analyzing the column predicted and expressions and generating pushdown.

That's why I put `connectorExpressionRewriter` in the `MongoSession` class, not in the `MongoMetadata` class.

 Do you think it is right?